src/composables/map/street-view.composable.ts

```diff
@@ -0,0 +1,255 @@
+import { Select } from 'ol/interaction'
+import { fromLonLat, toLonLat } from 'ol/proj'
+import { containsCoordinate } from 'ol/extent'
+import { loadGoogleapis } from '@/services/info/street-view'
+import { SvCompassFeature } from '@/services/info/sv-compass-feature'
+import { SvDirectionFeature } from '@/services/info/sv-direction-feature'
+
+import { Feature } from 'ol'
+import { Point } from 'ol/geom'
+import VectorLayer from 'ol/layer/Vector'
+import VectorSource from 'ol/source/Vector'
+
+export const DEFAULT_INFO_ZINDEX = 1500
+export const SV_FEATURE_LAYER_TYPE = 'svFeatureLayer'
+
+const SV_RADIUS = 90
+
+export default function useStreeView() {
+  const { infoOpen } = storeToRefs(useAppStore())
+  const {
+    locationInfo,
+    hidePointer,
+    isStreetviewActive,
+    noDataAtLocation,
+    panoPositionChanging,
+    svFeature,
+  } = storeToRefs(useInfoStore())
+
+  const map = useMap().getOlMap()
+
+  let googleService: any = null
+  let streetViewService: any = null
+  let panorama: any = null
+  let panoramaLinksListener: any = null
+  let panoramaPovListener: any = null
+  const selectFeature = new Select({
+    filter: (feature /*, layer*/) => feature instanceof SvDirectionFeature,
+  })
+  selectFeature.on('select', handleNavigate)
+
+  const svPoint = new Point([0, 0])
+  const svCompassFeature = new SvCompassFeature()
+  svCompassFeature.setGeometry(svPoint)
+
+  const svFeatureLayer = new VectorLayer({
+    source: new VectorSource({
+      features: [] as Feature[],
+    }),
+    // altitudeMode: 'clampToGround',
+    zIndex: DEFAULT_INFO_ZINDEX,
+  })
+  svFeatureLayer.set('cyLayerType', SV_FEATURE_LAYER_TYPE)
+  map.addLayer(svFeatureLayer)
+
+  function setLocation(loc: Coordinate | undefined) {
+    if (loc && panorama !== null && !panoPositionChanging.value) {
+      const lonlat = toLonLat(loc, map.getView().getProjection())
+      streetViewService.getPanorama(
+        {
+          location: {
+            lat: lonlat[1],
+            lng: lonlat[0],
+          },
+          radius: SV_RADIUS,
+        },
+        updatePanorama
+      )
+    }
+  }
+
+  function updatePanorama(data: any, status: any) {
+    if (status === googleService?.maps.StreetViewStatus.OK) {
+      noDataAtLocation.value = false
+      nextTick(() => {
+        panorama.setPosition(data.location.latLng)
+        panorama.setVisible(true)
+      })
+    } else {
+      noDataAtLocation.value = true
+      svFeature.value = undefined
+      panorama.setVisible(false)
+    }
+  }
+
+  watch(infoOpen, open => {
+    if (open) {
+      if (isStreetviewActive.value && locationInfo.value) {
+        setSvFeatures(locationInfo.value)
+      }
+    } else {
+      svFeature.value = undefined
+    }
+  })
+
+  watch(locationInfo, loc => {
+    if (loc) {
+      if (isStreetviewActive.value && !panoPositionChanging.value) {
+        setLocation(loc)
+      }
+    } else {
+      svFeature.value = undefined
+      map.getViewport().style.cursor = ''
```

Comment:
so where else do you set map cursor style? not in another watcher apparently, this is very difficult to find.

Reply:
added comment